config/initializers/rack_attack.rb

```diff
@@ -0,0 +1,213 @@
+
+    result
+  end
+
+  def anonymous?
+    !authenticated?
+  end
+
+  def authenticated?
+    # If we explicitly added a parameter to avoid updating last_request_at, honor it
+    return false if env['QUERY_STRING'].include?('skip_trackable=true')
+
+    rack_session = env['rack.session']
+    return false unless rack_session
+
+    # Check session variable to check if the user is logged in. This avoids db queries
+    rack_session['warden.user.user.key'].present?
+  end
+
+  protected
+
+  def internal_lb_check?
+    env['HTTP_USER_AGENT'] == 'ELB-HealthChecker/2.0' && path.include?('status') && trusted_proxy?
+  end
+
+  # is the source ip on a local or private network?
+  def trusted_proxy?
+    ActionDispatch::RemoteIp::TRUSTED_PROXIES.any? do |range|
+      range.include?(request_ip)
+    end
+  end
+end
+
+class Rack::Attack::Request
+  include RackAttackRequestHelpers
+end
+
+# rubocop:disable Style/IfUnlessModifier
+Rack::Attack.tap do |config|
+  # Throttling configuration
+  # * Multiple throttles can match the same request
+  # * Names must be unique or they will be silently overwritten
+
+  brute_force_options = {
+    limit: 20,
+    period: 3.minutes,
+  }
+
+  # Goal: prevent brute-force enumeration of credentials
+  config.throttle('per-ip limit on auth requests', **brute_force_options) do |request|
+    if request.tracking_enabled? && request.anonymous?
+      request.request_ip if request.authentication_attempt?
+    end
+  end
+
+  # Goal: prevent brute-force enumeration of password reset tokens
+  config.throttle('per-ip limit on password resets', **brute_force_options) do |request|
+    if request.tracking_enabled? && request.anonymous? && request.password_reset_attempt?
+      request.request_ip
+    end
+  end
+
+  # Goal: limit burst requests from unauthenticated clients, such as spiders
+  config.throttle(
+    'General per-ip limit on unauthenticated requests',
+    limit: 10,
+    period: 1.second,
+  ) do |request|
+    if request.tracking_enabled? && request.anonymous?
+      # there's a more restrictive throttle on the pdf
+      request.request_ip
+    end
+  end
+
+  # Goal: prevent scripts run through authenticated user accounts from harvesting data or excessive use of the site, with an allowance for known poor behavior for cohorts, roll-ups, and other pages that load more than one request per page.
+  config.throttle(
+    'General per-ip limit on authenticated requests',
+    limit: 150,
+    period: 5.seconds,
+  ) do |request|
+    if request.tracking_enabled? && request.authenticated?
+      request.request_ip
+    end
+  end
+
+  # Goal: Prevent excessive creation of search queries
+  config.throttle(
+    'per-ip limit on search query creation',
+    limit: 30,
+    period: 1.minute,
+  ) do |request|
+    if request.tracking_enabled? && request.authenticated? && request.post? && request.path == '/client_search_queries'
```

Comment:
Try and catch the other *search_queries" routes
```suggestion
    if request.tracking_enabled? && request.authenticated? && request.post? && request.path =~ /_?client_search_queries\z/
```